src/Sql/dbo/Stored Procedures/Organization_EnableCollectionEnhancements.sql

```diff
@@ -0,0 +1,160 @@
+                    UPDATE CG
+                    SET
+                        CG.[ReadOnly] = 0,
+                        CG.[HidePasswords] = 0,
+                        CG.[Manage] = 0
+                    FROM [dbo].[CollectionGroup] CG
+                    INNER JOIN [dbo].[Collection] C ON CG.[CollectionId] = C.[Id]
+                    INNER JOIN #TempStep1 TG ON CG.[GroupId] = TG.[GroupId]
+                    WHERE C.[OrganizationId] = TG.[OrganizationId];
+
+                    -- Insert new rows into [dbo].[CollectionGroup]
+                    INSERT INTO [dbo].[CollectionGroup] ([CollectionId], [GroupId], [ReadOnly], [HidePasswords], [Manage])
+                    SELECT C.[Id], TG.[GroupId], 0, 0, 0
+                    FROM [dbo].[Collection] C
+                    INNER JOIN #TempStep1 TG ON C.[OrganizationId] = TG.[OrganizationId]
+                    LEFT JOIN [dbo].[CollectionGroup] CG ON CG.[CollectionId] = C.[Id] AND CG.[GroupId] = TG.[GroupId]
+                    WHERE CG.[CollectionId] IS NULL;
+
+                    -- Update Group to clear AccessAll flag
+                    UPDATE G
+                    SET [AccessAll] = 0
+                    FROM [dbo].[Group] G
+                    INNER JOIN #TempStep1 TG ON G.[Id] = TG.[GroupId];
+
+                -- Step 2
+                    -- Update existing rows in [dbo].[CollectionUser]
+                    UPDATE target
+                    SET
+                        target.[ReadOnly] = 0,
+                        target.[HidePasswords] = 0,
+                        target.[Manage] = 0
+                        FROM [dbo].[CollectionUser] AS target
+                        INNER JOIN (
+                            SELECT C.[Id] AS [CollectionId], T.[OrganizationUserId]
+                            FROM [dbo].[Collection] C
+                            INNER JOIN #TempStep2 T ON C.[OrganizationId] = T.[OrganizationId]
+                        ) AS source
+                    ON target.[CollectionId] = source.[CollectionId] AND target.[OrganizationUserId] = source.[OrganizationUserId];
+
+                    -- Insert new rows into [dbo].[CollectionUser]
+                    INSERT INTO [dbo].[CollectionUser] ([CollectionId], [OrganizationUserId], [ReadOnly], [HidePasswords], [Manage])
+                    SELECT source.[CollectionId], source.[OrganizationUserId], 0, 0, 0
+                    FROM (
+                             SELECT C.[Id] AS [CollectionId], T.[OrganizationUserId]
+                             FROM [dbo].[Collection] C
+                                 INNER JOIN #TempStep2 T ON C.[OrganizationId] = T.[OrganizationId]
+                         ) AS source
+                    LEFT JOIN [dbo].[CollectionUser] AS target
+                        ON target.[CollectionId] = source.[CollectionId] AND target.[OrganizationUserId] = source.[OrganizationUserId]
+                    WHERE target.[CollectionId] IS NULL;
```

Comment:
💭 Why doesn't this follow the same pattern/structure as Step 1? I thought it would be pretty similar because it's the same thing but for orgUsers instead of Groups. However, this uses target/source naming and has nested SELECT queries. Is there some difference here that I'm missing or is this just a result of removing the batching logic?

Reply:
That makes sense and its easier to read. I have rewritten those parts here, what do you think?

```
UPDATE target
SET
    target.[ReadOnly] = 0,
    target.[HidePasswords] = 0,
    target.[Manage] = 0
FROM [dbo].[CollectionUser] AS target
INNER JOIN [dbo].[Collection] AS C ON target.[CollectionId] = C.[Id]
INNER JOIN #TempUsersAccessAll AS TU ON C.[OrganizationId] = TU.[OrganizationId] AND target.[OrganizationUserId] = TU.[OrganizationUserId];
```

```
INSERT INTO [dbo].[CollectionUser] ([CollectionId], [OrganizationUserId], [ReadOnly], [HidePasswords], [Manage])
SELECT C.[Id] AS [CollectionId], TU.[OrganizationUserId], 0, 0, 0
FROM [dbo].[Collection] C
INNER JOIN #TempUsersAccessAll TU ON C.[OrganizationId] = TU.[OrganizationId]
LEFT JOIN [dbo].[CollectionUser] target
    ON target.[CollectionId] = C.[Id] AND target.[OrganizationUserId] = TU.[OrganizationUserId]
WHERE target.[CollectionId] IS NULL;
```

@rkac-bw will perform some testing to see if there isn't any performance loss.

Reply:
LGTM